lib/internal/repl.js

```diff
@@ -51,15 +58,29 @@ function createRepl(env, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY_FILE) {
-    return setupHistory(repl, env.NODE_REPL_HISTORY_FILE, cb);
+  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
+    return setupHistory(repl, env.NODE_REPL_HISTORY,
+                        env.NODE_REPL_HISTORY_FILE, cb);
```

Comment:
@Fishrock123 @rvagg I think the bug is here. We call `setupHistory` with four arguments, but it only takes three.